python/cudf/cudf/tests/test_repr.py

```diff
@@ -84,34 +92,23 @@ def test_full_series(nrows, dtype):
     pd.reset_option("display.max_rows")
 
 
+@pytest.mark.parametrize("nrows", [5, 10, 15])
+@pytest.mark.parametrize("ncols", [5, 10, 15])
+@pytest.mark.parametrize("size", [20, 21])
 @pytest.mark.parametrize("dtype", repr_categories)
-@pytest.mark.parametrize("nrows", [0, 1, 2, 9, 20 / 2, 11, 20 - 1, 20, 20 + 1])
-@pytest.mark.parametrize("ncols", [0, 1, 2, 9, 20 / 2, 11, 20 - 1, 20, 20 + 1])
-def test_full_dataframe_20(dtype, nrows, ncols):
-    size = 20
+def test_full_dataframe_20(dtype, size, nrows, ncols):
     pdf = pd.DataFrame(
         {idx: np.random.randint(0, 100, size) for idx in range(size)}
     ).astype(dtype)
     gdf = cudf.from_pandas(pdf)
 
+    pd.options.display.max_rows = int(nrows)
+    pd.options.display.max_columns = int(ncols)
```

Comment:
Let's use an option context manager here instead of running `pd.reset_option` at the end. It's easy to forget and leave the global state altered by mistake, which caused failures for me when running subsets of the doctests. Example usage:

https://github.com/rapidsai/cudf/blob/2e458b93b639d5068452c9559ff045b60cc9b933/python/cudf/cudf/tests/test_dataframe.py#L753

The context manager should surround only the first assert statement (which should read `repr(pdf) == repr(gdf)`). I believe the other fancy repr methods are unaffected by those options. Then the `pd.reset_option` calls can be removed.

Reply:
> The context manager should surround only the first assert statement

I thought about this some more and I’m not sure if I was correct or not. Please check to see if the fancy repr calls are affected by the display options and include them in the context manager if so.

Reply:
I found all 3 are affected here.